study1/week2/longest-palindrome.java

```diff
@@ -0,0 +1,38 @@
+//https://leetcode.com/problems/longest-palindrome/description/
+
+class Solution {
+    public int longestPalindrome(String s) {
+        //Palindrome -> reads the same backwards as forwards
+        //how to check? -> if reverse string is same as original string
+      
+        //thennnn
+
+        //what rule? -> (even number of alphabets) or (even number of alphabets + only one odd number of alphabets) -> 2n or 2n+1
+        //Then check how many each letter is -> use count value
+        //If there is an even number of words in the string, remove them from the hashset, and then count + 1 each, then return count*2 when all are deleted (no alpabet in hashSet anymore)
+        //After repeating, there are odd numbers left at the end? -> return count*2 + 1
+        //Put s into the hashset, and count+1 if it already exists, then delete it, so i can figure out when there are odd numbers
+
+        //so time complexity is O(n)
+        //space complexity is also O(n)
+
+        int count = 0;
+        HashSet<Character> charSet = new HashSet<>();
```

Comment:
(responding to your PR description) You either need to use an array or a hashmap, so it's still okay to use hashmaps here. Using array will be slightly better in the number of computation, but it's still same complexity wise.